app/main.cpp

```diff
@@ -144,40 +110,59 @@ int main()
     const unsigned int vertexBufferSize = TriangleShapeVertices.size() * sizeof(float);
     GL_EXEC(glBufferData(GL_ARRAY_BUFFER, vertexBufferSize, TriangleShapeVertices.data(), GL_STATIC_DRAW));
 
+    // 3 generate element buffer object, load into OpenGL to use
+    unsigned int ebo;
+    GL_EXEC(glGenBuffers(1, &ebo));
+    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, ebo);
+    glBufferData(GL_ELEMENT_ARRAY_BUFFER, TriangleIndices.size() * sizeof(unsigned int), TriangleIndices.data(), GL_STATIC_DRAW);
+
     {
-        const auto vertexProgram = LoadFileString("Shaders\\mvp.vert");
-        const auto fragmentProgram = LoadFileString("Shaders\\ndc.frag");
+        const auto vertexProgram = LoadFileString("Shaders\\TextureMVP.vert");
+        const auto fragmentProgram = LoadFileString("Shaders\\TextureMVP.frag");
+        {
+            auto texture = Texture("Textures\\coordinate.jpg");
+            const auto textureId = texture.Load(OnLoadTexture);
+            glActiveTexture(GL_TEXTURE0);
+            GL_EXEC(glBindTexture(GL_TEXTURE_2D, textureId));
+        }
```

Comment:
1. Load texture as char array from image
2. load texture char array into gpu and return texture id
3. activate and bind texture id
4. exit scope, free texture memory